### PR TITLE
Apache and Project name changes

### PR DESCRIPTION
**Pull request by lbustelo**
Collection of changes due to new home and rename

- [x] Update project module names (for publishing to maven central)
- [x] Update project version
- [x] Update package names to be org.apache.toree
- [x] Update docs to reflect the name toree and update code samples to use new
namespaces
- [x] Update license headers to remove any spark kernel/IBM references

### CODE REVIEW

**Comment by lbustelo** [comment]
A couple of diffs that might need special attention


What is `spark-kernel` here?
```
--- a/kernel-api/src/main/scala/org/apache/toree/dependencies/IvyDependencyDownloader.scala
+++ b/kernel-api/src/main/scala/org/apache/toree/dependencies/IvyDependencyDownloader.scala
@@ -14,7 +14,7 @@
  * limitations under the License.
  */
 
-package com.ibm.spark.dependencies
+package org.apache.toree.dependencies
 
 import java.io.{File, PrintStream}
 import java.net.URL
@@ -87,7 +87,7 @@ class IvyDependencyDownloader(repositoryUrl: String, baseDirectory: String)
     ivyFile.deleteOnExit()
 
     val md = DefaultModuleDescriptor.newDefaultInstance(
-      ModuleRevisionId.newInstance("com.ibm.spark", "spark-kernel", "working")
+      ModuleRevisionId.newInstance("org.apache.toree", "spark-kernel", "working")
     )


```

What do we do about 3rd party magics and the package? (i.e. Brunel Magic)
```

--- a/kernel-api/src/main/scala/org/apache/toree/magic/MagicLoader.scala
+++ b/kernel-api/src/main/scala/org/apache/toree/magic/MagicLoader.scala
@@ -14,12 +14,12 @@
  * limitations under the License.
  */
 
-package com.ibm.spark.magic
+package org.apache.toree.magic
 
 import java.net.{URL, URLClassLoader}
 
 import com.google.common.reflect.ClassPath
-import com.ibm.spark.magic.dependencies.DependencyMap
+import org.apache.toree.magic.dependencies.DependencyMap
 
 import scala.reflect.runtime.{universe => runtimeUniverse}
 import scala.collection.JavaConversions._
@@ -29,7 +29,7 @@ class MagicLoader(
   urls: Array[URL] = Array(),
   parentLoader: ClassLoader = null
 ) extends URLClassLoader(urls, parentLoader) {
-  private val magicPackage = "com.ibm.spark.magic.builtin"
+  private val magicPackage = "org.apache.toree.magic.builtin"
 
   /**
    * Checks whether a magic with a given name, implementing a given interface,
diff --git a/kernel-api/src/main/scala/org/apache/toree/magic/dependencies/DependencyMap.scala b/kernel-api/src/main/scala/org/apache/toree/magic/dependencies/DependencyMap.scala
index f641a50..1f36e2b 100644


```

File: kernel-api/src/main/scala/org/apache/toree/dependencies/IvyDependencyDownloader.scala [hunk 1]
```diff
@@ -87,7 +88,7 @@ class IvyDependencyDownloader(repositoryUrl: String, baseDirectory: String)
     ivyFile.deleteOnExit()
 
     val md = DefaultModuleDescriptor.newDefaultInstance(
-      ModuleRevisionId.newInstance("com.ibm.spark", "spark-kernel", "working")
+      ModuleRevisionId.newInstance("org.apache.toree", "spark-kernel", "working")
```
Comment (chipsenkbeil) [inline kernel-api/src/main/scala/org/apache/toree/dependencies/IvyDependencyDownloader.scala]: This shouldn't be a problem. We are generating a new XML file that Ivy will use and had to create an overall name for it (org, artifact name, version, etc). Hence the `working` bit of it. You can change it to `toree-kernel` to be consistent if you want.

File: kernel-api/src/main/scala/org/apache/toree/magic/MagicLoader.scala [hunk 2]
```diff
@@ -29,7 +30,7 @@ class MagicLoader(
   urls: Array[URL] = Array(),
   parentLoader: ClassLoader = null
 ) extends URLClassLoader(urls, parentLoader) {
-  private val magicPackage = "com.ibm.spark.magic.builtin"
+  private val magicPackage = "org.apache.toree.magic.builtin"
```
Comment (chipsenkbeil) [inline kernel-api/src/main/scala/org/apache/toree/magic/MagicLoader.scala]: Ideally, we need a better system for detecting plugins (and support more kinds of plugins). I started working on this awhile ago using https://github.com/bmc/classutil to detect classes via annotations/interfaces. It uses the ASM bytecode library underneath, so the classutil library performs pretty well (and can work with jars that are not yet loaded).

File: kernel-api/src/main/scala/org/apache/toree/dependencies/IvyDependencyDownloader.scala (hunk 1 above)
Comment (lbustelo) [inline kernel-api/src/main/scala/org/apache/toree/dependencies/IvyDependencyDownloader.scala]: Should we rename the `spark-kernel` string to just `kernel`?

Comment (chipsenkbeil) [inline kernel-api/src/main/scala/org/apache/toree/dependencies/IvyDependencyDownloader.scala]: Sure.

**Comment by lbustelo** [comment]
Fixes https://issues.apache.org/jira/browse/TOREE-247

**Comment by lbustelo** [comment]
We've completed all the Todo's for this PR. This is ready to merge.

**Pull request merged**